### ISSUE TITLE
[Feature Request] Autoremove videos from Watch later after watching!

### ISSUE THREAD

**Issue by Alwinator** [yuliskov/SmartTube#328]
It would be extremely convenient, especially because it takes forever to remove every video by hand.

And by the way, thank you a lot for your great app!

**Comment by Alwinator** [comment]
@yuliskov Is this feature already planned for one of the future versions?

**Comment by Daniel15** [comment]
I'd also love to see this! I started using this app when the official YouTube app started behaving weirdly for me, and it works perfectly! Would love to see this feature though.

**Comment by Alwinator** [comment]
Is there any news? @yuliskov @foreveryoungforty

**Comment by elle8che** [comment]
See 1173 and https://github.com/yuliskov/SmartTubeNext/issues/1173#issuecomment-1050078541

**Comment by Alwinator** [comment]
> See 1173 and https://github.com/yuliskov/SmartTubeNext/issues/1173#issuecomment-1050078541

Thank you for referencing, but in my opinion that is not the same. The remove all watched videos feature would be very risky, so I agree with comments in the other issue. However, if you just remove one video after watching it is not risky and extremely convenient.

**Comment by elle8che** [comment]
> > See 1173 and [#1173 (comment)](https://github.com/yuliskov/SmartTubeNext/issues/1173#issuecomment-1050078541)
> 
> Thank you for the reference, but in my opinion, #1173 is not the same. The remove all watched videos feature would be very risky, so I agree with the comments in the other issue. However, if you just remove one video after watching it, it is not risky and extremely convenient.

You're absolutely right. I read too fast. I agree -- removing them as we watch them would be a nice option.

**Comment by PapeCoding** [comment]
I would also be very interested in this feature!

**Comment by matheus-partschi** [comment]
This feature would be amazing.

**Comment by matheusfaustino** [comment]
This would an amazing feature. Just found out about the playlist limit from YouTube, it is tedious to remove all videos or multiple videos from youtube.

**Comment by ACR-Jeff** [comment]
This feature has been implemented in current latest releases
`Settings > General > Hide content > Hide watched videos from Watch later playlist`

**Comment by Daniel15** [comment]
Are they actually removed, or just hidden? This request was for removing them.

**Comment by ACR-Jeff** [comment]
> Are they actually removed, or just hidden? This request was for removing them.
> […](#)

If they are hidden then they are removed correct? You don't see them in the playlist after they are hidden, I don't understand the debate, Maybe we can get some information to clarify this from @yuliskov

**Comment by Daniel15** [comment]
Playlists have a limit of 5000 videos, which is why I'd like videos to be removed after watching them. I hit the 5000 limit and had to make a "watch later 2" playlist, since removing them one by one is a hassle.

**Comment by ACR-Jeff** [comment]
That's very understandable, Thank you for updating me on this issue, This helped me better understand the issue, I hope we can get some more information on this.
